### PR TITLE
feat: first steps for making jax masked implementation even more accessible.

### PR DESCRIPTION
**Pull request by Solosneros**
Hi @sebasrb09 @jjalko ,

I split the functions in separate files and remove unused functions. Furthermore, I renamed some variables and removed the explicit profiling imports.

Could you please have a look? Thanks!

### CODE REVIEW

File: src/data.py
Comment (jjalko) [inline src/data.py:1]: maybe we could remove the device calls after the numpy loads, as I don't think this is supported for the np arrays.

File: src/dp_accounting_utils.py
Comment (jjalko) [inline src/dp_accounting_utils.py:1]: Are we defining the `compute_epsilon` to verify the privacy cost at the end of the run? If so, then it might give us a too large epsilon in case one uses "prv" accounting for the `calculate_noise` function.

File: src/example.py [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+
+def _parse_arguments(args, dataset_size):
+    num_steps = args.epochs * math.ceil(dataset_size / args.bs)
+
+    q = 1 / math.ceil(dataset_size / args.bs)
+
+    noise_std = calculate_noise(q, args.epsilon, args.target_delta, args.epochs, args.accountant)
+    C = args.grad_norm
+
+    optimizer_config = namedtuple("Config", ["momentum", "learning_rate"])
```
Comment (jjalko) [inline src/example.py:38]: Since we are using `optax.adam`, the momentum parameter is not required. Furthermore, it's not passed to the optimizer later on in the `example.py`.

Comment (Solosneros) [reply src/example.py]: I removed it now, only leaving the learning_rate. Is passing this named_tuple kinda the default in JAX?

File: src/data.py
Comment (Solosneros) [reply src/data.py:1]: True, I removed them now.

File: src/dp_accounting_utils.py
Comment (Solosneros) [reply src/dp_accounting_utils.py:1]: You are right, that might happen and that is the intended reason for the function.

I was thinking that we could leave the accounting for the next iteration of improvements once we know which accountant to use. I only left both `compute_epsilon` and `calculate_noise` so that we have functions with the correct signature in the correct places.

File: src/example.py (hunk 1 above)
Comment (jjalko) [reply src/example.py]: I don't think it is a default. It's just for convenience of separating the optimizer parameters into a separate object. You might as well pass a dictionary, and change how the values are accessed.
